bin/bank/pycbc_brute_bank

```diff
@@ -247,20 +249,52 @@ class TriangleBank(object):
     def check_params(self, gen, params, threshold):
         num_tried = 0
         num_added = 0
-        for i in range(len(tuple(params.values())[0])):
-            num_tried += 1.0
-
-            try:
-                hp = gen.generate(**{key:params[key][i] for key in params})
-            except Exception as err:
-                print(err)
-                continue
-
-            hp.gen = gen
-            hp.threshold = threshold
-            if hp not in self:
-                num_added += 1
-                self.insert(hp)
+        total_num = len(tuple(params.values())[0])
+
+        if args.num_cores == 1:
```

Comment:
@yi-fan-wang Use the pycbc pool object. You won't need this if statement then and so you can combine the two redundant sections here. I think it might be ok to just always cache?

Reply:
@ahnitz implemented as you said. Seems to work in a test for both single core (nprocess=1) and multiple cores. But I'm not familiar with pycbc.pool so could you help me double check?